TypeCobol/Compiler/Sql/Model/Information.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using TypeCobol.Compiler.CodeElements;
+
+namespace TypeCobol.Compiler.Sql.Model
+{
+    public abstract class Information : SqlObject
+    {
+        public abstract InformationType Type { get; }
+        protected override bool VisitSqlObject(ISqlVisitor visitor)
+        {
+            return visitor.Visit(this);
+        }
+    }
+
+    public enum InformationType
+    {
+        Statement,
+        Condition,
+        Combined
+    }
+
+    public class StatementInformation : Information
+    {
+        public StatementInformation(List<InformationAssignment> assignments)
+        {
+            Assignments = assignments;
+        }
+
+        public override InformationType Type => InformationType.Statement;
+
+        public List<InformationAssignment> Assignments { get; }
+
+        protected override void DumpContent(TextWriter output, int indentLevel)
+        {
+            DumpProperty(output, nameof(Assignments), Assignments, indentLevel);
+        }
+
+        protected override bool VisitSqlObject(ISqlVisitor visitor)
+        {
+            return base.VisitSqlObject(visitor) && visitor.Visit(this) &&
+                   visitor.ContinueVisit(Assignments);
+        }
+    }
+
+    public abstract class InformationAssignment : SqlObject
+    {
+        protected InformationAssignment(SqlVariable storage)
+        {
+            Storage = storage;
+        }
+
+        public SqlVariable Storage { get; }
+
+        protected override void DumpContent(TextWriter output, int indentLevel)
+        {
+            DumpProperty(output, nameof(Storage), Storage, indentLevel);
+        }
+
+        protected override bool VisitSqlObject(ISqlVisitor visitor)
+        {
+            return visitor.Visit(this) &&
+                   visitor.ContinueVisit(Storage);
+        }
+    }
+
+    public class SingleInformationAssignment : InformationAssignment
+    {
+        public SingleInformationAssignment(SymbolReference itemName, SqlVariable storage) : base(storage)
+        {
+            ItemName = itemName;
+        }
+
+        public SymbolReference ItemName { get; }
+
+        protected override void DumpContent(TextWriter output, int indentLevel)
+        {
+            base.DumpContent(output, indentLevel);
+            DumpProperty(output, nameof(ItemName), ItemName, indentLevel);
+        }
+
+        protected override bool VisitSqlObject(ISqlVisitor visitor)
+        {
+            return base.VisitSqlObject(visitor) && visitor.Visit(this);
+        }
+    }
+
+    public class CompositeInformationAssignment : InformationAssignment
+    {
+        public CompositeInformationAssignment(List<SymbolReference> itemNames, SqlVariable storage) : base(storage)
```

Comment:
Swap param order

Reply:
CompositeInformationAssignment class is now merged into InformationAssignment.